apps/studio.giselles.ai/app/(main)/settings/account/actions.ts

```diff
@@ -141,3 +154,126 @@ export async function leaveTeam(
 	}
 	return result;
 }
+
+function getExtensionFromMimeType(mimeType: string): string {
+	return (
+		IMAGE_CONSTRAINTS.mimeToExt[
+			mimeType as keyof typeof IMAGE_CONSTRAINTS.mimeToExt
+		] || "jpg"
+	);
+}
+
+/**
+ * Gets a cached avatar URL, either returning an external URL directly or generating a signed URL for Supabase storage
+ *
+ * @param path - The path to the avatar image (full URL for external sources, storage path for Supabase)
+ * @returns A URL that can be used to display the avatar
+ *
+ * Note on timing:
+ * - Signed URL is set to expire in 1 hour (3600 seconds)
+ * - Cache is set to revalidate in 55 minutes (3300 seconds)
+ * This 5-minute buffer ensures we generate a new signed URL before the current one expires
+ */
+const getAvatarUrl = unstable_cache(
+	async (path: string): Promise<string> => {
+		if (path.startsWith("https://")) {
+			return path;
+		}
+
+		try {
+			const signedUrl = await storage.getItem(path, {
+				signedUrl: true,
+				expiresIn: 60 * 60, // 1 hour expiry for signed URL
+			});
+
+			if (!signedUrl || typeof signedUrl !== "string") {
+				throw new Error("Failed to generate signed URL");
+			}
+
+			return signedUrl;
+		} catch (error) {
+			logger.error("Failed to generate signed URL:", error);
+			throw new Error("Failed to generate avatar URL");
+		}
+	},
+	["avatar-url"],
+	{
+		revalidate: 55 * 60, // Revalidate cache after 55 minutes
+		tags: ["avatar-url"],
+	},
+);
```

Comment:
I implemented signed URLs to access images from a private bucket. However, this implementation might not be optimal if other team members need to access avatars as well.

A better approach would be to:
1. Create a new public bucket specifically for avatars
2. Store public URLs from Supabase Storage, similar to how we handle social provider avatars
3. Consider implementing custom domain settings for better control and branding

This would improve performance and simplify the implementation while maintaining consistent behavior across different avatar sources.